### DISCUSSION TITLE
Is it correct that adding the FmodBankLoader as an autoloaded Scene increases performance?

### DISCUSSION THREAD

**Discussion by 3ter** [utopia-rise/fmod-gdextension#222] [Q&A] [answered]
I wasn't sure if I understood the docs (https://fmod-gdextension.readthedocs.io/en/latest/user-guide/3-nodes/) correctly here. I added a scene with an `FmodBankLoader` in it, where I add all the banks we're using in the project and set this scene to be on Autoload.
![image](https://github.com/utopia-rise/fmod-gdextension/assets/11499929/456a9132-7d0a-499e-bd33-2666d00b9ecf)

I understand that when I now add an `FmodEventEmitter`, add an event to it, it is going to be loaded from the `FmodBankLoader` without any additional loading which is for performance reasons.

Is there anything else to it, or am I doing it wrong 😆 ?

**Comment by coletdementa** [comment, marked as answer]
Hey there!

It looks like you’re using the autoloaded BankLoader to ensure that the master, strings, and music banks are always available, that’s great! However, I noticed you’re also keeping the SFX bank loaded all the time. While that can work fine, it really depends on your project’s needs.

For example, in a small arcade-style game where the same SFX are used across menus and gameplay (like player attacks or enemy deaths), keeping the SFX bank loaded globally makes sense. But in larger projects with many unique or scene-specific sounds, it might be more efficient to load those SFX banks only when needed.

You could, for instance, give your Player scene a dedicated Loader node that loads the Player SFX bank. That way, if there’s no Player in the current scene, you’re not keeping its sound data in memory unnecessarily.

This approach helps keep things organized and can offer a small performance boost, though of course, it depends on the overall sound scope of your project.
